tests/js/realm-tests.js

```diff
@@ -325,6 +332,16 @@ module.exports = BaseTest.extend({
                 realm.create('InvalidObject', {intCol: 1});
             });
         });
+
+        // Only the original constructor should be valid.
+        function WrongCustomObject() {}
+        WrongCustomObject.schema = CustomObject.schema;
```

Comment:
Since we don't retain that object, the user is free to do whatever they want to it from my perspective. It would be weird to freeze it.